### PR TITLE
Custom LED implementation (Web Config)

### PR DESCRIPTION
**Pull request by FeralAI**
Addresses https://github.com/OpenStickCommunity/GP2040-CE/issues/107

This is the web UI for the custom LED configuration. It's been added to a new section of the existing LED configuration page. Here are the functional details:

* User is able select from the 3 existing LED layouts and have the preview adjust accordingly
* User can then click on any controller button to bring up the color picker for normal and pressed states
* User can right-click on a button to see the pressed state
* Option to set all buttons to a single color at once
* Option to clear all colors, with confirmation modal

I also fixed a couple warnings in the React build from the recent Keyboard mode impelmentation due to loose equality checks.

### CODE REVIEW

**Comment by FeralAI** [comment]
I'm currently discussing some minor styling updates with @TheTrainGoes. Will add a comment when this is finalized.

**Comment by FeralAI** [comment]
Ok this is ready to go 👍

I ran into a bug when exporting a backup. It seems there was a limited buffer or silent error and only part of my export was being written. The export logic was refactored in [94ebdb2](https://github.com/OpenStickCommunity/WebConfigurator/pull/38/commits/94ebdb27ac4b9b66cfb4c80dc8bf1bcc562d6e88) to use `Blob` and `URL.createObjectURL` instead of manually composing the file data string.

File: src/Pages/KeyboardMapping.js [hunk 1]
```diff
@@ -149,9 +149,9 @@ export default function KeyboardMappingPage() {
 					continue;
 
 				if (mappings[prop].key !== 0x00 && mappings[prop].key === mappings[otherProp].key)
-					mappings[prop].error = `Key ${KEY_CODES.find(({label, value}) => mappings[prop].key == value).label} is already assigned`;
+					mappings[prop].error = `Key ${KEY_CODES.find(({label, value}) => mappings[prop].key === value).label} is already assigned`;
```
Comment (FeralAI) [inline src/Pages/KeyboardMapping.js:152]: Fixes linter warning.

**Review by arntsonl — APPROVED** [review]
Looks great, some good little fixes in there as well.

**Pull request merged**